### ISSUE TITLE
Test Thyra endpoints

### ISSUE THREAD

**Issue by thomas-senechal** [massalabs/station#514]
**Context**
Thyra is mostly stable and must be tested to ensure that every features are working as expected. For that, we need to test Thyra's endpoints.

**User flow**
No user flow since it's a purely development oriented task

**How to**

- [x] List the parts of Thyra that needs to be tested
- [ ] Make sure all endpoints are well tested
    - [x] https://github.com/massalabs/thyra/issues/559
    - [x] https://github.com/massalabs/thyra/issues/560
    - [x] https://github.com/massalabs/thyra/issues/561
    - [x] https://github.com/massalabs/thyra/issues/562
    - [x] https://github.com/massalabs/thyra/issues/585
    - [ ] WebOnChain: 
        - [x] https://github.com/massalabs/thyra/issues/600
        - [ ] https://github.com/massalabs/thyra/issues/601
        - [x] https://github.com/massalabs/thyra/issues/613
        - [ ] https://github.com/massalabs/thyra/issues/614
- [ ] Make sure to have some "scenarios" that will match how we use the endpoints in Thyra

**Technical details**

- Create a `.robot` test file for each parts that needs to be tested
- Write tests for each part of Thyra in the corresponding `.robot` test file
- Once every related issues are done:
    - [ ] deleted KPI as discussed in [this comment](https://github.com/massalabs/thyra/issues/514#issuecomment-1460188271) 
    - [ ] remove tests from the Postman Collection
    - [x] remove Postman tests from the CI https://github.com/massalabs/thyra/pull/714

**QA testing**
- TBD (Please contact me if not updated so we can discuss about the best way to QA review it)
- Check that the latest CI is green

<details>
<summary>Previous version</summary>

> We wanted to use Postman as our test framework. But it clearly doesn't matches our needs as explained in [this comment](https://github.com/massalabs/thyra/issues/514#issuecomment-1468926066).

**Context**
Thyra is mostly stable and must be tested to ensure that every features are working as expected. For that, we need to add tests to Thyra's Postman collection. 

**How to**
- List the parts of Thyra that needs to be tested
- Add missing endpoints to the Postman collection
- Add tests for all endpoints in the Postman collection
- Make sure all endpoints are well tested

</details>

**Comment by thomas-senechal** [comment]
I just remembered about this issue: https://github.com/massalabs/thyra/issues/504 which should be done once https://github.com/massalabs/thyra/issues/441 is done. This could allow us to do the tests for the `executeFunction` and the Event Manager since we will be able to deploy SC easily.

**Comment by SlnPons** [comment]
* Wallet: legacy code. It will be deprecated and so endpoint tests will be added on the wallet plugin side. Inno team will sync with Run on this subject when ready.

* KPI: don't know what it is. @gregLibert would you know? 

* Plugin manager: it's a priority to test them all. 

* I also have endpoints that you haven't mentioned, see below: 
<img width="739" alt="image" src="https://user-images.githubusercontent.com/109611779/223666989-8143663d-d88b-468d-88e9-6e1ca264285c.png">
--> if this is what you call 'front-end' then it needs to be tested and added as a priority because the end-to-end tests about the web on-chain won't happen soon. Not a focus on our side atm. 

* Web on-chain: we should break this down to identify what's linked to the storage vs. Dns and other related thyra's pkg so that we can anticipate the work needed to test them as well as to isolate them and so test them after. 

* executeFunction: deploySC and CallSC are related to this endpoint (i think) and so this is critical endpoint to test for thyra. I'm not sure why you can't see it in used.... maybe we should talk about it. 

* event endpoint: same pluggenisation is still not in the pipe so keen to understand the amount of work on this one so we can decide with which project we start (pluggenisation or test).

In conclusion, could you add more details about web on-chain endpoints, even, front-end and executeFunction ones please? 
My objective is to be able to evaluate if we should add test now or pluggenise now. 
If you're also able to evaluate the work to pluggenise the related object to web on-chain & event manager that would be even better.

**Comment by thomas-senechal** [comment]
> --> if this is what you call 'front-end' then it needs to be tested and added as a priority because the end-to-end tests about the web on-chain won't happen soon. Not a focus on our side atm.

Yes, I was talking about those endpoints as "front-end". At least for the ones ending with `resource`. 
We will test them then 

> * Web on-chain: we should break this down to identify what's linked to the storage vs. Dns and other related thyra's pkg so that we can anticipate the work needed to test them as well as to isolate them and so test them after.

- `/browse/{address}/{resource}` -> Storage
- `/all/domains/` -> DNS
- `/my/domains/{nickname}` -> DNS
- `/websiteCreator/prepare/` -> DNS
- `/websiteCreator/upload/` -> none
- `/websiteCreator/uploadMissingChunks/` -> none


> * executeFunction: deploySC and CallSC are related to this endpoint (i think) and so this is critical endpoint to test for thyra. 

Yes. `CallSC` is the Operation sent by `/cmd/executeFunction` endpoint.

> I'm not sure why you can't see it in used.... maybe we should talk about it.

AFAIK, right now, we don't `/cmd/executeFunction` endpoint within Thyra front end. 

> * event endpoint: same pluggenisation is still not in the pipe so keen to understand the amount of work on this one so we can decide with which project we start (pluggenisation or test).

The event manager needs a SC that pushes events on the blockchain to be tested. I think it should be possible to test it once https://github.com/massalabs/thyra/issues/441 is merged.

**Comment by gregLibert** [comment]
> KPI: don't know what it is. @gregLibert would you know?

It was my very first demo of thyra API if I remember well. To be deleted :-)

**Comment by SlnPons** [comment]
@thomas-senechal after discussing with the team on my side, it seems way better to secure the endpoint testing before isolating the objects so that you will be able to isolate and see what breaks. 

So I would suggest to take all thyra endpoints that exist today and add test on all of them. 
Once all tests exist, we will be able to isolate the objects securely and without stress :) 

wdyt?

**Comment by thomas-senechal** [comment]
After discussing with @gregLibert , we decided to stop using Postman for testing due to its limitations. For example, Postman doesn't allow us to really control the order of the tests, we cannot make multiple requests in one test, some devs need to modify tests to adapt tests to their environment... To name a few limitations.

We talked about using a scripting language such as Python or JS to right tests, the issue is that this limits the addition of tests to the devs. To open test writing to QAs and PMs, we should use a Language Agnostic framework.

Gregory proposed to use RobotFramework. I made some researches but couldn't find another well known Language Agnostic test framework. That's why I decided to start trying to make tests with RobotFramework.

**Comment by SlnPons** [comment]
Super! Let's go with this solution for now then 💪

**Comment by gregLibert** [comment]
Mostly done. I propose to reopen an issue if needed.
